### ISSUE TITLE
xmlschema.exceptions.XMLSchemaParseError: duplicated identity constraint

### ISSUE THREAD

**Issue by Thomasb81** [brunato/xmlschema#19]
Hello
I try to use this schema definition : http://www.eda.org/xmlschema/spirit/1685-2009
but it failed with rising an exception:

```
    raise error
xmlschema.exceptions.XMLSchemaParseError: duplicated identity constraint u'{http://www.spiritconsortium.org/XMLSchema/SPIRIT/1685-2009}registerFileRegisterFieldKey':

  <xs:key xmlns:xs="http://www.w3.org/2001/XMLSchema" name="registerFileRegisterFieldKey">
                    <xs:selector xpath="spirit:field" />
                    <xs:field xpath="spirit:name" />
  </xs:key>
```
Look like there is several node that lead to record what you call "constraint".

My case:
the schema [1685-2009.zip](https://github.com/brunato/xmlschema/files/1363428/1685-2009.zip)
```
import xmlschema
my_schema = xmlschema.XMLSchema('1685-2009/component.xsd')
```

Thomas

**Comment by Thomasb81** [comment]
More interesting when the archive contain files: [1685-2009.zip](https://github.com/brunato/xmlschema/files/1363857/1685-2009.zip)

**Comment by brunato** [comment]
The word "constraint" refers to XML Schema Identity constraints, a feature added in the version 0.9.12.

In the version 0.9.13 the error you reported is fixed but another error persists:
```
  File "/home/brunato/.virtualenvs/xmlschema3/lib/python3.6/site-packages/xmlschema/components/component.py", line 106, in _parse_error
    raise error
xmlschema.exceptions.XMLSchemaParseError: empty context fields for <XsdKeyref '{http://www.spiritconsortium.org/XMLSchema/SPIRIT/1685-2009}componentChoiceRef' at 0x7f890fbd4710>:

  <xs:keyref xmlns:xs="http://www.w3.org/2001/XMLSchema" name="componentChoiceRef" refer="spirit:componentChoiceKey">
            <xs:selector xpath=".//*" />
            <xs:field xpath="@spirit:choiceRef" />
  </xs:keyref>
```

A partial rewrite of identity constraint's code is needed (_set_context_ method) because descendant paths (//) doesn't work well with schema components's structure. Maybe it will be done for the next version.

Meanwhile you could use the schema using the _lax_ validation mode:

```
>>> my_schema = xmlschema.XMLSchema('1685-2009/component.xsd', validation='lax')
>>> my_schema.all_errors
[XMLSchemaParseError("empty context fields for <XsdKeyref '{http://www.spiritconsortium.org/XMLSchema/SPIRIT/1685-2009}componentChoiceRef' at 0x7f890f850b70>:", <XsdKeyref '{http://www.spiritconsortium.org/XMLSchema/SPIRIT/1685-2009}componentChoiceRef' at 0x7f890f850b70>, <Element '{http://www.w3.org/2001/XMLSchema}keyref' at 0x7f890fa5ce58>)]

```

**Comment by brunato** [comment]
This bug persists in the version 0.9.14, published today.
Still postponed to a future version.

**Comment by brunato** [comment]
Should be fixed in version 0.9.17 (tested with an XML instance of the _1685-2009_ schema).
